llvm/lib/Target/PowerPC/PPCISelLowering.cpp

```diff
@@ -11479,9 +11492,33 @@ SDValue PPCTargetLowering::LowerSCALAR_TO_VECTOR(SDValue Op,
   EVT PtrVT = getPointerTy(DAG.getDataLayout());
   SDValue FIdx = DAG.getFrameIndex(FrameIdx, PtrVT);
 
+  SDValue Val = Op.getOperand(0);
+  EVT ValVT = Val.getValueType();
+  // P10 hardware store forwarding requires that a single store contains all
+  // the data for the load. P10 is able to merge a pair of adjacent stores. Try
```

Comment:
For `adjacent stores` in P10 hardware store forwarding, do you mean there should be no instructions between these two stores in the final assembly? I tend to "no" as we have OOO on P10. But if the answer is yes, maybe we need a pseudo instruction to pack these two stores together and expand them after the schedulers... The chain in the codes can only guarantee the first store is before the second store.

Reply:
Adjacent here means that the storage targets modified by the two stores have to be adjacent in memory. Other instructions can be between them, but not other stores.

Reply:
Thanks for explanation. Got it.